engine/swarm/secrets.md

```diff
@@ -0,0 +1,339 @@
+`/run/secrets/<secret_name>`. You can update a service to grant it access to
+additional secrets or revoke its access to a given secret at any time.
+
+A node only has access to (encrypted) secrets if it is a swarm manager or if it
+is running service tasks which have been granted access to the secret. When a
+container task stops running, the decrypted secrets shared to it are unmounted
+from the in-memory filesystem for that container and flushed from the node's
+memory.
+
+If a node loses connectivity to the swarm while it is running a task container
+with access to a secret, the task container still has access to its secrets, but
+cannot receive updates until the node reconnects to the swarm.
+
+You can add or inspect an individual secret at any time, or list all
+secrets. You cannot remove a secret that a running service is
+using. See [Rotate a secret](secrets.md#example-rotate-a-secret) for a way to
+remove a secret without disrupting running services.
+
+In order to update or roll back secrets more easily, consider adding a version
+number or date to the secret name. This is made easier by the ability to control
+the mount point of the secret within a given container.
+
+## Read more about `docker secret` commands
+
+Use these links to read about specific commands, or continue to the
+[example about using secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+- [`docker secret create`](../reference/commandline/secret_create.md)
+- [`docker secret inspect`](../reference/commandline/secret_inspect.md)
+- [`docker service ls`](../reference/commandline/secret_ls.md)
+- [`docker secret rm`](../reference/commandline/secret_rm.md)
+- [`--secret`](../reference/commandline/service_create.md#create-a-service-with-secrets) flag for `docker service create`
+- [`--secret-add` and `--secret-rm`](../reference/commandline/service_update.md#adding-and-removing-secrets) flags for `docker service update`
+
+## Example: Get started with secrets
+
+This simple example shows how secrets work in just a few commands. For a
+real-world example, continue to
+[Use secrets with a service](secrets.md#example-use-secrets-with-a-service).
+
+1.  Add a secret to Docker. The `docker secret create` command reads standard
+    input.
+
+    ```bash
+    $ echo "This is a secret" | docker secret create secret_data.txt
+    ```
+
+2.  Create a `redis` service and grant it access to the secret.
+    ```bash
+    $ docker service  create --name="redis" --secret="secret_data.txt" redis:alpine
+    ```
+
+3.  Get the ID of the `redis` service task container using `docker ps`. The ID
+    is the first field.
+
+    ```bash
+    $ docker ps |grep redis
+
+    ea3839998cbf        redis@sha256:f1ed3708f538b537eb9c2a7dd50dc90a706f7debd7e1196c9264edeea521a86d                "docker-entrypoint..."   About a minute ago   Up About a minute   6379/tcp              redis.1.rn9sbes7lxsiaizfqzo7g4nvu
```

Comment:
This output will be updated before GA to match the 1.12 output again, and the digest will no longer be shown by default, just `redis:latest`